check.py

```diff
@@ -118,15 +158,16 @@ def check_repo(file_paths, print_all, white_listed_urls, white_listed_patterns,
     check_results = check_repo(file_paths, print_all, white_listed_urls,
                                white_listed_patterns, retry_count, timeout)
 
-    # delete repo when done
-    deletion_status = del_repo(base_path)
+    # delete repo when done, if requested
+    if cleanup == "true":
+        del_repo(base_path)
 
```

Comment:
I am not sure if I get this change.  
We usually clone the repo, run checks and the ideal final step is to clean up by removing the cloned repo so why would the user would want to keep the clone? can you please explain the logic behind it?

Reply:
Most users that use the action will get the repo by way of the github/actions/checkout step run before, in which case the repository isn't cloned, but the base path used is that checked out repo. 95% of users will not want the check to be run and then have their repository deleted. Let me know if this is clear or if you need clarification on any point.

Reply:
Here is an explicit, real world example. If I were to delete the repository after running the URL check step, none of the testing steps that follow would work.

```

    steps:
    - uses: actions/checkout@v2
    - name: URLs-checker
      uses: urlstechie/URLs-checker@add/checkout-optional-git-path
      with:
        subfolder: docs
    - name: Run Tests
      run: |
         # This (and anything following) will not work if URLs-checker deletes my repo!
         pytest tests/*.py
```